File: api/auth_config.go
package api

import (
	"fmt"
	"net/http"
	"strings"
	"time"

	"github.com/bugfan/ang-admin/models"
	"github.com/bugfan/ang-admin/service"
	"github.com/bugfan/rest"
	"github.com/gin-gonic/gin"
	"github.com/go-xorm/xorm"
)

func init() {
	rest.Register(&models.Auth{}, &authHandler{}, rest.RouteTypeALL, nil, "auth")
}

type authHandler struct {
	Id            int64     `json:"id"`
	Name          string    `json:"name"`
	AuthMethodIds string    `json:"auth_method_ids"`
	PortalUrl     string    `json:"portal_url"`
	Remark        string    `json:"remark"`
	CreatedAt     time.Time `json:"created_at"`
	UpdatedAt     time.Time `json:"updated_at"`
}

func (h *authHandler) Before(g *gin.Context, x *xorm.Engine) bool {
	method := g.Request.Method
	if method == http.MethodPost || method == http.MethodPut || method == http.MethodPatch {
		h.Name = strings.TrimSpace(h.Name)
		if h.Name == "" {
			g.AbortWithStatusJSON(http.StatusOK, gin.H{
				"code":      1,
				"error_key": "nameRequired",
				"message":   "认证配置名称不能为空",
			})
			return false
		}

		// Check name unique
		var exist models.Auth
		has, err := x.Where("name = ? AND id != ?", h.Name, h.Id).Get(&exist)
		if err == nil && has {
			g.AbortWithStatusJSON(http.StatusOK, gin.H{
				"code":      1,
				"error_key": "nameDuplicate",
				"message":   fmt.Sprintf("认证配置名称 [%s] 已存在", h.Name),
			})
			return false
		}
	} else if method == http.MethodDelete {
		// 这里可以加一个校验：如果该 auth 正在被 HTTP/TCP 等规则使用，则不能删除
		// 暂时先不做协议关联的校验，先实现基础 CRUD
	}
	return true
}

func (h *authHandler) After(g *gin.Context, x *xorm.Engine, args ...interface{}) {
	method := g.Request.Method
	if method == "POST" || method == "PUT" || method == "PATCH" || method == "DELETE" {
		service.SyncHTTPToCluster()
	}
}

File: web/src/api/auth-config.ts
import { http } from "@/utils/http";

export interface AuthItem {
  id?: number;
  name?: string;
  auth_method_ids?: string;
    portal_url?: string;
    remark?: string;
  created_at?: string;
}

export type AuthResult = {
  code: number;
  message: string;
  data: {
    list: Array<AuthItem>;
    total: number;
  };
};

export const getAuthList = async (params?: object) => {
  try {
    const res = await http.request<any>("get", "/api/auth", { params });
    if (Array.isArray(res)) {
      return {
        code: 0,
        message: "success",
        data: {
          list: res,
          total: res.length,
          pageSize: 10,
          currentPage: 1
        }
      };
    }
    if (res && typeof res === "object" && res.list) {
      return { code: 0, message: "success", data: res };
    }
    return {
      code: 0,
      message: "success",
      data: { list: [], total: 0, pageSize: 10, currentPage: 1 }
    };
  } catch (err: any) {
    return {
      code: 1,
      message: "获取列表失败",
      data: { list: [], total: 0, pageSize: 10, currentPage: 1 }
    };
  }
};

export const createAuth = (data: object) => {
  return http.request<any>("post", "/api/auth", { data }).then(res => {
    if (res && typeof res.code === "number" && res.code !== 0) return res;
    return { code: 0, message: "success", data: res };
  });
};

export const updateAuth = (id: number, data: object) => {
  return http.request<any>("put", `/api/auth/${id}`, { data }).then(res => {
    if (res && typeof res.code === "number" && res.code !== 0) return res;
    return { code: 0, message: "success", data: res };
  });
};

export const deleteAuth = (id: number) => {
  return http.request<any>("delete", `/api/auth/${id}`).then(res => {
    if (res && typeof res.code === "number" && res.code !== 0) return res;
    return { code: 0, message: "success", data: res };
  });
};
